guardian/actions-riff-raff
# feat: Add tags to uploaded objects

## What does this change?
Add the following tags to files uploaded to Riff-Raff's buckets:
- `gu:for-github-repository`, this matches the tag name from [riffraff-platform](https://github.com/guardian/riffraff-platform/blob/9f5e58defcd4f27906c28769062cf84b99fdceb1/packages/cdk/lib/roles-for-github.ts#L91-L94)
- `gu:for-riffraff-project`

This is to aid Riff-Raff enforcing a relation between repository, project, stack, and app.

Also included is some refactors to the S3 interactions, removing the `S3Store` class in favour of DI functions.

## How to test
Use this branch, and observe the tags on the resulting files in S3.

For example, the uploaded `riff-raff.yaml` file from https://github.com/guardian/service-catalogue/pull/861 has these tags:

```console
➜ aws s3api get-object-tagging --bucket riffraff-artifact --key deploy::service-catalogue/3982/riff-raff.yaml
{
  "TagSet": [
    {
      "Key": "gu:for-github-repository",
      "Value": "https://github.com/guardian/service-catalogue"
    },
    {
      "Key": "gu:for-riffraff-project",
      "Value": "deploy::service-catalogue"
    }
  ]
}
```
